### PR TITLE
Examples to glm

### PR DESCRIPTION
**Pull request by roymacdonald**
@arturoc

### CODE REVIEW

File: examples/3d/normalsExample/src/ofApp.cpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ void ofApp::draw(){
 
     ofSetColor(255);
     ofDrawBitmapString("press any key or mouse button to disable mesh normals", 20,20);
-    ofDrawBitmapString("light", cam.worldToScreen(light.getGlobalPosition()) + ofPoint(10,0));
+    ofDrawBitmapString("light", cam.worldToScreen(light.getGlobalPosition()) + glm::vec3(20,0,0));
```
Comment (bakercp) [inline examples/3d/normalsExample/src/ofApp.cpp]: Any reason why 10 changed to 20?

File: examples/3d/quaternionArcballExample/src/ofApp.cpp [hunk 2]
```diff
@@ -26,17 +26,18 @@ void ofApp::update(){
 
 //--------------------------------------------------------------
 void ofApp::draw(){
-
+
```
Comment (bakercp) [inline examples/3d/quaternionArcballExample/src/ofApp.cpp]: Space cleanup.

File: examples/android/androidAdvanced3DExample/src/Swarm.cpp [hunk 3]
```diff
@@ -45,8 +45,8 @@ void Swarm::init(int _nParticles, float positionDispersion, float velocityDisper
 	// SETUP ARRAYS
 	///////////////////////////////////////////
 	//
-	positions = new ofVec3f[nParticles];
```
Comment (bakercp) [inline examples/android/androidAdvanced3DExample/src/Swarm.cpp]: This may be a different project, but we should get rid of heap allocation here and instead use `std::vector<>`

File: examples/android/androidGuiExample/Makefile [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Attempt to load a config.make file.
```
Comment (bakercp) [inline examples/android/androidGuiExample/Makefile]: Do we want to add these project files?

File: examples/android/androidGuiExample/src/ofApp.cpp [hunk 5]
```diff
@@ -27,7 +27,7 @@ void ofApp::setup(){
 	
 	bHide = true;
 
-	ring.loadSound("ring.wav");
```
Comment (bakercp) [inline examples/android/androidGuiExample/src/ofApp.cpp]: Outside the scope of the PR, but looks good anyway.

File: examples/android/androidVibratorExample/src/ofApp.cpp [hunk 6]
```diff
@@ -54,11 +54,11 @@ void ofApp::touchDown(int x, int y, int id){
 void ofApp::touchMoved(int x, int y, int id){
 	bool found=false;
 	for(int i=0;i<areas.size();i++){
-		if(areas[i].inside(ofVec2f(x,y))){
+		if(areas[i].inside(x,y)){
 			found=true;
 			break;
 		}
-	}
+	}ss
```
Comment (bakercp) [inline examples/android/androidVibratorExample/src/ofApp.cpp]: `ss`?

File: examples/gl/billboardRotationExample/src/ofApp.cpp [hunk 7]
```diff
@@ -42,13 +42,14 @@ void ofApp::setup() {
 
 //--------------------------------------------------------------
 void ofApp::update() {
-	ofVec2f mouse(ofGetMouseX(), ofGetMouseY());
-	ofVec2f mouseVec(ofGetPreviousMouseX()-ofGetMouseX(), ofGetPreviousMouseY()-ofGetMouseY());
-	mouseVec.limit(10.0);
+	glm::vec2 mouse(ofGetMouseX(), ofGetMouseY());
+	glm::vec2 mouseVec(ofGetPreviousMouseX()-ofGetMouseX(), ofGetPreviousMouseY()-ofGetMouseY());
+	glm::clamp(mouseVec, 0.0f, 10.0f);
+//	mouseVec.limit(10.0);
```
Comment (bakercp) [inline examples/gl/billboardRotationExample/src/ofApp.cpp]: Remove commented line.

File: examples/gl/computeShaderParticlesExample/src/ofApp.cpp [hunk 8]
```diff
@@ -54,15 +54,15 @@ void ofApp::update(){
...
-			ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.2),0,1,-ofGetHeight()*size,ofGetHeight()*size),
-			ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.5),0,1,0,-ofGetHeight()*size));
-	atractor2.set(ofMap(ofNoise(ofGetElapsedTimef()*0.5+0.3),0,1,-ofGetWidth()*size,ofGetWidth()*size),
-			ofMap(ofNoise(ofGetElapsedTimef()*0.5+0.2),0,1,-ofGetHeight()*size,ofGetHeight()*size),
-			ofMap(ofNoise(ofGetElapsedTimef()*0.5+0.1),0,1,0,-ofGetHeight()*size));
-	atractor3.set(ofMap(ofNoise(ofGetElapsedTimef()*0.9+0.1),0,1,-ofGetWidth()*size,ofGetWidth()*size),
-			ofMap(ofNoise(ofGetElapsedTimef()*0.9+0.5),0,1,-ofGetHeight()*size,ofGetHeight()*size),
-			ofMap(ofNoise(ofGetElapsedTimef()*0.9+0.7),0,1,0,-ofGetHeight()*size));
+	atractor1 = {ofMap(ofNoise(ofGetElapsedTimef()*0.3),0,1,-ofGetWidth()*size,ofGetWidth()*size),
+				 ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.2),0,1,-ofGetHeight()*size,ofGetHeight()*size),
```
Comment (bakercp) [inline examples/gl/computeShaderParticlesExample/src/ofApp.cpp]: Odd spacing?

File: examples/gl/fboTrailsExample/src/ofApp.cpp [hunk 9]
```diff
@@ -118,16 +118,16 @@ void ofApp::draw(){
...
+    ofDrawBitmapString("floating point FBO", 420,20);
 
 	string alphaInfo = "Current alpha fade amnt = " + ofToString(fadeAmnt);
 	alphaInfo += "\nHold '1' to set alpha fade to 1";
 	alphaInfo += "\nHold '2' to set alpha fade to 5";
 	alphaInfo += "\nHold '3' to set alpha fade to 15";
 	alphaInfo += "\nHold 'c' to clear the fbo each frame\n\nMove mouse to draw with a circle";
 
-    ofDrawBitmapString(alphaInfo, ofPoint(10,430));
+    ofDrawBitmapString(alphaInfo, 10,430);
```
Comment (bakercp) [inline examples/gl/fboTrailsExample/src/ofApp.cpp]: Odd comma spacing.

File: examples/3d/normalsExample/src/ofApp.cpp (hunk 1 above)
Comment (roymacdonald) [reply examples/3d/normalsExample/src/ofApp.cpp]: not sure, did this a long ago, probably it looks better???

File: examples/gl/glInfoExample/src/ofApp.cpp [hunk 10]
```diff
@@ -236,8 +236,8 @@ void ofApp::draw(){
 
 
 
-    ofDrawBitmapStringHighlight(output, ofPoint(20,20));
-    ofDrawBitmapStringHighlight("press ' ' to load full report", ofPoint(20,220), ofColor::magenta, ofColor::white);
+    ofDrawBitmapStringHighlight(output, 20,20);
+    ofDrawBitmapStringHighlight("press ' ' to load full report", 20,220, ofColor::magenta, ofColor::white);
```
Comment (bakercp) [inline examples/gl/glInfoExample/src/ofApp.cpp]: More odd spacing.

File: examples/gl/gpuParticleSystemExample/src/ofApp.cpp [hunk 11]
```diff
@@ -80,8 +80,8 @@ void ofApp::setup(){
     mesh.setMode(OF_PRIMITIVE_POINTS);
     for(int x = 0; x < textureRes; x++){
         for(int y = 0; y < textureRes; y++){
-            mesh.addVertex(ofVec3f(x,y));
-            mesh.addTexCoord(ofVec2f(x, y));
+			mesh.addVertex({x,y,0});
+			mesh.addTexCoord({x, y});
         }
```
Comment (bakercp) [inline examples/gl/gpuParticleSystemExample/src/ofApp.cpp]: Odd spacing.

File: examples/gl/pixelBufferExample/src/ofApp.cpp [hunk 12]
```diff
@@ -10,15 +10,17 @@ void ofApp::setup(){
...
 	record = false;
 }
 
 //--------------------------------------------------------------
 void ofApp::update(){
-	box.setOrientation(ofQuaternion(45, ofVec3f(1,0,0),
-			45,ofVec3f(0,0,1),
-			ofGetElapsedTimef()*10,	ofVec3f(0,1,0)));
+	box.setOrientation(glm::angleAxis(ofDegToRad(ofGetElapsedTimef()*10),	glm::vec3(0,1,0)) *//y axis rotation
+					   glm::angleAxis(45.0f, glm::vec3(0,0,1))*//z axis rotation
```
Comment (bakercp) [inline examples/gl/pixelBufferExample/src/ofApp.cpp]: This is strangely formatted with with comments ... it appears that it should be multiplied, but should probably be pulled out to a local variable for clarity.

File: examples/gl/threadedPixelBufferExample/src/ofApp.cpp [hunk 13]
```diff
@@ -10,16 +10,16 @@ void ofApp::setup(){
...
 	firstFrame = true;
 }
 
 //--------------------------------------------------------------
 void ofApp::update(){
-	box.setOrientation(ofQuaternion(45, ofVec3f(1,0,0),
-			45,ofVec3f(0,0,1),
-			ofGetElapsedTimef()*10,	ofVec3f(0,1,0)));
+	box.setOrientation(glm::angleAxis(ofDegToRad(ofGetElapsedTimef()*10),	glm::vec3(0,1,0))*
+					   glm::angleAxis(45.0f,glm::vec3(0,0,1))*
```
Comment (bakercp) [inline examples/gl/threadedPixelBufferExample/src/ofApp.cpp]: More spacing changes.

File: examples/gl/vboExample/src/ofApp.cpp [hunk 14]
```diff
@@ -12,27 +12,27 @@
...
-void ofApp::addFace(ofMesh& mesh, ofVec3f a, ofVec3f b, ofVec3f c) {
+void ofApp::addFace(ofMesh& mesh, const glm::vec3& a, const glm::vec3& b, const glm::vec3& c) {
 	mesh.addVertex(a);
 	mesh.addVertex(b);
 	mesh.addVertex(c);
 }
 
 //--------------------------------------------------------------
-void ofApp::addFace(ofMesh& mesh, ofVec3f a, ofVec3f b, ofVec3f c, ofVec3f d) {
+void ofApp::addFace(ofMesh& mesh, const glm::vec3& a, const glm::vec3& b, const glm::vec3& c, const glm::vec3& d) {
```
Comment (bakercp) [inline examples/gl/vboExample/src/ofApp.cpp]: Thank you for `const` :)

File: examples/math/trigonometryExample/src/ofApp.cpp [hunk 15]
```diff
@@ -35,27 +35,26 @@ void ofApp::setup(){
...
     
     if (!ofGetMousePressed()) {//press the mouse to stop the angle from incrementing.
 		angle+=rotationSpeed;//at each update the angle get's incremented 
 	}
-	if (angle>=TWO_PI) { //if the angle is more than or equal to two PI (a full rotation measured in Radians) then make it zero.
-		angle=0;
-	}
-
+	 //if the angle is more than or equal to two PI (a full rotation measured in Radians) then make it zero.
+	angle = ofWrap(angle, 0, TWO_PI);
```
Comment (bakercp) [inline examples/math/trigonometryExample/src/ofApp.cpp]: This is not an equivalent replacement.

Comment (bakercp) [reply examples/math/trigonometryExample/src/ofApp.cpp]: ofWrap Wrapping will wrap around zero, not clamp.

File: examples/math/trigonometryExample/src/ofApp.cpp [hunk 16]
```diff
@@ -146,27 +145,25 @@ void ofApp::keyReleased(int key){
 void ofApp::mouseMoved(int x, int y ){
 
 }
-
+//--------------------------------------------------------------
+void ofApp::checkMouse(float x, float y){
```
Comment (bakercp) [inline examples/math/trigonometryExample/src/ofApp.cpp]: Is this new method just to reduce repetitive code?

File: examples/math/vectorMathExample/src/ofApp.cpp [hunk 17]
```diff
@@ -136,10 +136,10 @@ void ofApp::keyPressed  (int key){
 
     switch (key){
         case 'a':
-            speedOfRotation += 0.05f;
+            speedOfRotation += 0.001f;
```
Comment (bakercp) [inline examples/math/vectorMathExample/src/ofApp.cpp]: Why this change?

File: examples/math/trigonometryExample/src/ofApp.cpp (hunk 15 above)
Comment (bakercp) [reply examples/math/trigonometryExample/src/ofApp.cpp]:
```
/// \brief Find a value within a given range, wrapping the value if it overflows.
///
/// If a value is between from and to, return that value.
/// If a value is NOT within that range, wrap it.
///
/// Example:
/// ~~~~{.cpp}
///		ofWrap(5, 0, 10); // Returns 5.
///		ofWrap(15, 0, 10); // Also returns 5.
///		ofWrap(-5, 0, 10); // Also returns 5.
/// ~~~~
///
/// \param value The value to map.
/// \param from The lower bound of the range.
/// \returns to The upper bound of the range.
```

File: examples/3d/quaternionArcballExample/src/ofApp.cpp [hunk 18]
```diff
@@ -58,20 +59,20 @@ void ofApp::mouseMoved(int x, int y ){
...
-
+	
 	//every time the mouse is dragged, track the change
 	//accumulate the changes inside of curRot through multiplication
-    ofVec2f mouse(x,y);
-    ofQuaternion yRot((x-lastMouse.x)*dampen, ofVec3f(0,1,0));
-    ofQuaternion xRot((y-lastMouse.y)*dampen, ofVec3f(-1,0,0));
-    curRot *= yRot*xRot;
+	glm::vec2 mouse(x,y);
+	glm::quat yRot = glm::angleAxis(ofDegToRad(x-lastMouse.x)*dampen, glm::vec3(0,1,0));
```
[the remark this answers is not in the document]
Comment (roymacdonald) [reply examples/3d/quaternionArcballExample/src/ofApp.cpp]: I would say the OF functions as these are named a bit more explicitly.

File: examples/android/androidGuiExample/Makefile (hunk 4 above)
Comment (roymacdonald) [reply examples/android/androidGuiExample/Makefile]: I think that these should not be there as no example carries a makefile. I'll remove these

File: examples/android/androidVibratorExample/src/ofApp.cpp (hunk 6 above)
Comment (roymacdonald) [reply examples/android/androidVibratorExample/src/ofApp.cpp]: typo. sorry

File: examples/math/trigonometryExample/src/ofApp.cpp (hunk 15 above)
Comment (roymacdonald) [reply examples/math/trigonometryExample/src/ofApp.cpp]: and the original was warping not clamping but only in one direction

File: examples/math/trigonometryExample/src/ofApp.cpp (hunk 16 above)
Comment (roymacdonald) [reply examples/math/trigonometryExample/src/ofApp.cpp]: yes.

File: examples/3d/normalsExample/src/ofApp.cpp (hunk 1 above)
Comment (roymacdonald) [reply examples/3d/normalsExample/src/ofApp.cpp]: just checked. otherwise it overlaps with the sphere and can not be read properly

File: examples/math/vectorMathExample/src/ofApp.cpp (hunk 17 above)
Comment (roymacdonald) [reply examples/math/vectorMathExample/src/ofApp.cpp]: otherwise the change is too abrupt. it feels better I think

File: examples/gl/pixelBufferExample/src/ofApp.cpp [hunk 19]
```diff
@@ -16,9 +16,9 @@ void ofApp::setup(){
 
 //--------------------------------------------------------------
 void ofApp::update(){
-	box.setOrientation(glm::angleAxis(ofDegToRad(ofGetElapsedTimef()*10),	glm::vec3(0,1,0)) *//y axis rotation
```
[the remark this answers is not in the document]
Comment (bakercp) [reply examples/gl/pixelBufferExample/src/ofApp.cpp]: Fix here https://github.com/roymacdonald/openFrameworks/pull/2/commits/eb84f4a719e4621843d4d77d067405bfe510c7a3

File: examples/gl/computeShaderParticlesExample/src/ofApp.cpp [hunk 20]
```diff
@@ -55,14 +55,14 @@ void ofApp::update(){
 	compute.setUniform1f("elapsedTime",ofGetElapsedTimef());
 	float size = 4;
 	atractor1 = {ofMap(ofNoise(ofGetElapsedTimef()*0.3),0,1,-ofGetWidth()*size,ofGetWidth()*size),
-				 ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.2),0,1,-ofGetHeight()*size,ofGetHeight()*size),
-				 ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.5),0,1,0,-ofGetHeight()*size)};
+	             ofMap(ofNoise(ofGetElapsedTimef()*0.3+0.2),0,1,-ofGetHeight()*size,ofGetHeight()*size),
```
[the remark this answers is not in the document]
Comment (bakercp) [reply examples/gl/computeShaderParticlesExample/src/ofApp.cpp]: Fix for this is in PR.

File: examples/sound/soundPlayerExample/bin/data/sounds/ofApp.cpp [hunk 21]
```diff
@@ -0,0 +1,180 @@
+#include "ofApp.h"
```
[the remark this answers is not in the document]
Comment (roymacdonald) [reply examples/sound/soundPlayerExample/bin/data/sounds/ofApp.cpp]: lol. my fault. don't know how this got here.

File: examples/gl/pixelBufferExample/src/ofApp.cpp (hunk 19 above)
[the remark this answers is not in the document]
Comment (roymacdonald) [reply examples/gl/pixelBufferExample/src/ofApp.cpp]: thanks. I remembered that this worked, though this looks a lot more easier to understand.

File: examples/gl/computeShaderParticlesExample/src/ofApp.cpp (hunk 20 above)
[the remark this answers is not in the document]
Comment (bakercp) [reply examples/gl/computeShaderParticlesExample/src/ofApp.cpp]: Fixed.

File: examples/sound/soundPlayerExample/bin/data/sounds/ofApp.cpp (hunk 21 above)
[the remark this answers is not in the document]
Comment (bakercp) [reply examples/sound/soundPlayerExample/bin/data/sounds/ofApp.cpp]: Fixed.
